### PR TITLE
Folder Structure / Sudoer / admin user updates with ban-ip.sh

### PR DESCRIPTION
**Pull request by DanVanAtta**
A number of changes included, please see the commit comments for details. 

The important part of the folder structure update is that we no longer have a 'system_update.sh', everything is now done from 'host_control.sh', and the installations are now grouped between `roles/application` and `roles/system`

After the folder update commit the infrastructure project looks like this:
screenshot from 2018-10-27 15-17-15

Also of note is a change to how `/etc/sudoer` is managed and we'll explicitly set it in each server instead of using grep-fu. As part of the updated file is the copy from warclub that gives admin full sudo (so we will not have to worry about managing /etc/sudoer as much going forward).

Last addition is a ban-ip.sh script. The intent is that we'll have system-admins and @prastle use that to firewall block a given IP.

### CODE REVIEW

File: update_cron.sh [hunk 1]
```diff
@@ -28,7 +28,8 @@ if [ -z "$CONTROL_FILE" ]; then
 
   echo "Cron executing ${date} ${hostname}" > "${LOG_FILE}"
   chown -R admin:admin "${LOG_FOLDER}"
-  /root/infrastructure/system_update.sh 1>> "${LOG_FILE}" | /usr/bin/logger
```
Comment (DanVanAtta) [inline update_cron.sh]: This is part of the simplification/folder update; `system_update.sh` essentially set a path variable and then called "system_control.sh" and then "host_control.sh". Now that we only need "host_control.sh", we can flatten this and invoke host_control.sh directly.

File: roles/system/admin_user/files/block-ip.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  echo "This script will firewall block the given ip address parameter"
+  exit 1
+}
+
+if [ -z "${IP_TO_BLOCK}" ]; then
+  usage
+fi
+
+echo "$(date) - Blocking IP: ${IP_TO_BLOCK}" >> /home/admin/ip-blocks.log
+sudo ufw deny from ${IP_TO_BLOCK} to any
```
Comment (ssoloff) [inline roles/system/admin_user/files/block-ip.sh]: I know on the production lobby server, there are already numerous `ALLOW` rules in place.  I believe this command will add the new `DENY` rule to the end of the list.  If there is an `ALLOW` rule that matches before the `DENY`, the `DENY` will never get hit.

You may have to always insert it at the top of the list using `ufw insert 1 deny ...` to ensure the specific `DENY` rules get processed before the generic `ALLOW` rules.

File: roles/system/admin_user/files/block-ip.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+set -eux
```
Comment (ssoloff) [inline roles/system/admin_user/files/block-ip.sh]: Did you want to leave `-x` in here?  Or was it just for debugging?

File: roles/system/admin_user/files/block-ip.sh (hunk 2 above)
Comment (DanVanAtta) [reply roles/system/admin_user/files/block-ip.sh]: I did not know that behavior, thanks - updated in: 6a81774

File: roles/system/admin_user/files/block-ip.sh (hunk 3 above)
Comment (DanVanAtta) [reply roles/system/admin_user/files/block-ip.sh]: Probably was not intentional, can't rightly remember, I removed the 'x' flag: a861c7e

**Comment by DanVanAtta** [comment]
@RoiEXLab the complete lock-out is intended. An IP ban is to get those that have otherwise defeated our other bans

**Pull request merged**